activesupport/lib/active_support/cache.rb

```diff
@@ -843,49 +842,60 @@ def [](version)
       module Loader
         extend self
 
+        def dump(entry)
+          serialize(entry)
+        end
+
+        def dump_compressed(entry, threshold)
+          serialize(entry.compressed(threshold))
+        end
+
         def load(payload)
           if !payload.is_a?(String)
             ActiveSupport::Cache::Store.logger&.warn %{Payload wasn't a string, was #{payload.class.name} - couldn't unmarshal, so returning nil."}
 
             return nil
           elsif payload.start_with?(MARK_70_UNCOMPRESSED)
-            members = Marshal.load(payload.byteslice(1..-1))
+            members = loaded(payload.byteslice(1..-1))
           elsif payload.start_with?(MARK_70_COMPRESSED)
-            members = Marshal.load(Zlib::Inflate.inflate(payload.byteslice(1..-1)))
+            members = loaded(Zlib::Inflate.inflate(payload.byteslice(1..-1)))
           elsif payload.start_with?(MARK_61)
-            return Marshal.load(payload)
+            return loaded(payload)
           else
             ActiveSupport::Cache::Store.logger&.warn %{Invalid cache prefix: #{payload.byteslice(0).inspect}, expected "\\x00" or "\\x01"}
 
             return nil
           end
           Entry.unpack(members)
         end
+
+        private
+          def loaded(payload)
+            Marshal.load(payload)
+          rescue TypeError
+            JSON.decode(payload, symbolize_names: true)
+          end
+
+          def serialize(payload)
+            Marshal.dump(payload)
+          end
```

Comment:
Is there a reason why we can't name these `serialize` and `deserialize` ?

Reply:
Sound's good!

Reply:
Thinking about this more, what situation are we trying to account for by falling back to JSON given a `TypeError` in the default `loaded` ?

We're overwriting it in the`Rails70Coder` right ?